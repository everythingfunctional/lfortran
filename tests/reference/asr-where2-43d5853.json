{
    "basename": "asr-where2-43d5853",
    "cmd": "lfortran --indent --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/where2.f90",
    "infile_hash": "4e202862988600ae3ddbfdd4c98cb526ed74a125d8285e9274d80b5a",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-where2-43d5853.stdout",
    "stdout_hash": "b966751a746eedb991f00a3435ad89271f8ad977d2eb474fd3ea480f",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}