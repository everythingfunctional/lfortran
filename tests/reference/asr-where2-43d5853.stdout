(TranslationUnit
    (SymbolTable
        1
        {
            main:
                (Program
                    (SymbolTable
                        3
                        {
                            
                        })
                    main
                    []
                    [(SubroutineCall
                        1 where2
                        ()
                        []
                        ()
                    )]
                ),
            where2:
                (Function
                    (SymbolTable
                        2
                        {
                            a:
                                (Variable
                                    2
                                    a
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4 [((IntegerConstant 1 (Integer 4 []))
                                    (IntegerConstant 10 (Integer 4 [])))])
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            b:
                                (Variable
                                    2
                                    b
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4 [((IntegerConstant 1 (Integer 4 []))
                                    (IntegerConstant 10 (Integer 4 [])))])
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    where2
                    (FunctionType
                        []
                        ()
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        []
                        []
                        .false.
                    )
                    []
                    []
                    [(Where
                        (IntegerCompare
                            (Var 2 a)
                            Gt
                            (Var 2 b)
                            (Logical 4 [((IntegerConstant 1 (Integer 4 []))
                            (IntegerConstant 10 (Integer 4 [])))])
                            ()
                        )
                        [(=
                            (Var 2 a)
                            (IntegerConstant 1 (Integer 4 []))
                            ()
                        )]
                        []
                    )
                    (Where
                        (IntegerCompare
                            (Var 2 a)
                            Eq
                            (IntegerConstant 1 (Integer 4 []))
                            (Logical 4 [((IntegerConstant 1 (Integer 4 []))
                            (IntegerConstant 10 (Integer 4 [])))])
                            ()
                        )
                        [(=
                            (Var 2 a)
                            (IntegerConstant 2 (Integer 4 []))
                            ()
                        )]
                        [(Where
                            (IntegerCompare
                                (Var 2 a)
                                Eq
                                (IntegerConstant 2 (Integer 4 []))
                                (Logical 4 [((IntegerConstant 1 (Integer 4 []))
                                (IntegerConstant 10 (Integer 4 [])))])
                                ()
                            )
                            [(=
                                (Var 2 a)
                                (IntegerConstant 3 (Integer 4 []))
                                ()
                            )]
                            [(=
                                (Var 2 a)
                                (IntegerBinOp
                                    (Var 2 a)
                                    Mul
                                    (IntegerConstant 2 (Integer 4 []))
                                    (Integer 4 [((IntegerConstant 1 (Integer 4 []))
                                    (IntegerConstant 10 (Integer 4 [])))])
                                    ()
                                )
                                ()
                            )]
                        )]
                    )]
                    ()
                    Public
                    .false.
                    .false.
                )
        })
    []
)
